src/vim.js

```diff
@@ -1808,17 +1810,43 @@ export function initVim(CodeMirror) {
             vimGlobalState.exCommandHistoryController.reset();
         }
       }
+      /**
+       * @arg {KeyboardEvent&{target:HTMLInputElement}} e
+       * @arg {any} query
+       */
+      function onPromptKeyUp(e, query) {
+        var inputStream = new CodeMirror.StringStream(query);
+        var params = {};
+        try {
+          exCommandDispatcher.parseInput_(cm, inputStream, params);
+          if (params.commandName != "s") {
+            clearSearchHighlight(cm);
+            return;
+          }
+          command = exCommandDispatcher.matchCommand_(params.commandName);
+          exCommandDispatcher.parseCommandArgs_(inputStream, params, command);
+          if (!params.argString) return;
+          var regex = parseQuery(params.argString.slice(1), true, true);
+          if (regex) highlightSearchMatches(cm, regex);
+        } catch(e) {
+          console.log(e);
```

Comment:
hmm not sure what we've been doing elsewhere but ideally we don't log too much to the user's console